### PR TITLE
Issue 148

### PR DESCRIPTION
**Pull request by jacobdotcosta**
Changes for a dynamic inventory from passwordstore projecet.

### CODE REVIEW

File: ansible/inventory/pass_inventory.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+result['_meta'] = {}
+result['_meta']['hostvars'] = {}
+# ansible_connection = 'passwordstore'
+password_store_dir = environ.get('PASSWORD_STORE_DIR')
+
+# print("password_store_dir: " + password_store_dir)
+
+f = []
+
+# def a(dirpath, arrdirnames, filenames): walk('/home/ajc102/dev/client/redhat-snowdrop/passstore/snowdrop')
```
Comment (cmoulliard) [inline ansible/inventory/pass_inventory.py]: Why do you hard coded your local path ?

File: ansible/inventory/pass_inventory.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+#! /usr/bin/python
+
```
Comment (cmoulliard) [inline ansible/inventory/pass_inventory.py]: Can you document what this python script is doing ?

File: ansible/playbook/k8s_extras.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+---
```
Comment (cmoulliard) [inline ansible/playbook/k8s_extras.yml]: Such playbooks `k8s.yml, k8s-misc.yml` already exist under the path `kubernetes/ansible/`. So why do you still use it here ?

File: ansible/inventory/pass_inventory.py (hunk 1 above)
Comment (jacobdotcosta) [reply ansible/inventory/pass_inventory.py]: It's commented out, I'll remove it.

File: ansible/playbook/README.md [hunk 4]
```diff
@@ -1,4 +1,98 @@
...
+When in this document it refers to the `controller` it means the machine that executes the Ansible playbooks and roles and that owns the inventory, *i.e.*, the 
+machine that clones this project.
+
+The `hosts` are the machines where the playbooks and roles are executed against. 
+
+The exception goes to the playbooks that are executed against `localhost`. This special host is referred to the `controller`. 
+
+# Ansible Inventory
+
+Ansible inventory manages access to hosts using Ansible.
```
Comment (cmoulliard) [inline ansible/playbook/README.md]: Is this sentence correct - `Ansible inventory manages access to hosts using Ansible` as it looks like for an ellipse ?

File: ansible/playbook/README.md [hunk 5]
```diff
@@ -1,4 +1,98 @@
...
+
+```bash
+$ ansible-playbook ansible/playbook/passstore_controller_inventory.yml -e vm_name=<vm_name> -e pass_provider=<provider> --tag "create"
+```
+
+The VM name is the name that the host will have in the inventory and the provider is the cloud provider. 
+
+> NOTE: ATTOW the only provider tested is `hetzner`. 
+
+## Remove a Ansible
```
Comment (cmoulliard) [inline ansible/playbook/README.md]: Is this title correct or should be renamed to `Remove a host` ?

File: ansible/playbook/README.md [hunk 6]
```diff
@@ -1,4 +1,98 @@
...
+```bash
+$ pass rm ansible/inventory/k8s_115/h01-k115
+```
+
+# Host management in Ansible
+
+The section describes how to maintain hosts in Ansible and their use. It's execution goes against the `controller`.
+
+## Create a host
+
```
Comment (cmoulliard) [inline ansible/playbook/README.md]: We should perhaps mention as pre-req to `git pull` the password project from gitlab - yes/no ? like also how we culd specify another password store path `export PASSWORD_STORE_DIR=~/.password-store-snowdrop` ?

**Comment by cmoulliard** [comment]
I did a test without success top of this branch
```
export vm_name=k03-k116
export PASSWORD_STORE_DIR=~/.password-store-snowdrop
ansible-playbook ansible/playbook/passstore_controller_inventory.yml -e vm_name=${vm_name} -e pass_provider=hetzner --tag "create"
...
TASK [passstore/ansible_inventory : Set pass facts for RSA keys] ***********************************************************************************************************************************************************************************
Tuesday 10 March 2020  18:57:18 +0100 (0:00:00.039)       0:00:18.172 ********* 
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'passwordstore'. Error was a <class 'IndexError'>, original message: list index out of range"} 
```

File: ansible/playbook/README.md (hunk 4 above)
Comment (jacobdotcosta) [reply ansible/playbook/README.md]: I improved the description.

File: ansible/playbook/README.md (hunk 6 above)
Comment (jacobdotcosta) [reply ansible/playbook/README.md]: That information is on gitlab. I'm not very fond of adding that information here as it is private. But I could add a small reference.

File: ansible/playbook/README.md (hunk 4 above)
Comment (jacobdotcosta) [reply ansible/playbook/README.md]: Done.

**Pull request merged**